File: GTTS/INL.TrainingService.Models/SaveTrainingEvent_Param.cs
using System;
using System.Collections.Generic;

namespace INL.TrainingService.Models
{
    public class SaveTrainingEvent_Param : ISaveTrainingEvent_Param
    {
		public Int64? TrainingEventID { get; set; }
		public string Name { get; set; }
		public string NameInLocalLang { get; set; }
		public int TrainingEventTypeID { get; set; }
        public string Justification { get; set; }
        public string Objectives { get; set; }
        public string ParticipantProfile { get; set; }
        public string SpecialRequirements { get; set; }
        public int? ProgramID { get; set; }
        public int? KeyActivityID { get; set; }
        public Int64 TrainingUnitID { get; set; }
        public int? CountryID { get; set; }
        public int? PostID { get; set; }
        public int? ConsularDistrictID { get; set; }
        public int? OrganizerAppUserID { get; set; }
        public int? PlannedParticipantCnt { get; set; }
        public int? PlannedMissionDirectHireCnt { get; set; }
        public int? PlannedNonMissionDirectHireCnt { get; set; }
        public int? PlannedMissionOutsourceCnt { get; set; }
        public int? PlannedOtherCnt { get; set; }
        public decimal? EstimatedBudget { get; set; }
        public decimal? ActualBudget { get; set; }
        public int? EstimatedStudents { get; set; }
        public int? FundingSourceID { get; set; }
        public int? AuthorizingLawID { get; set; }
        public string Comments { get; set; }
        public int? ModifiedByAppUserID { get; set; }

        public List<SaveTrainingEventLocation_Item> TrainingEventLocations { get; set; }
        public List<KeyActivity_Item> KeyActivities { get; set; }
        public List<IAA_Item> IAAs { get; set; }
        public List<SaveTrainingEventUSPartnerAgency_Item> TrainingEventUSPartnerAgencies { get; set; }
        public List<SaveTrainingEventProjectCode_Item> TrainingEventProjectCodes { get; set; }
        public List<SaveTrainingEventStakeholder_Item> TrainingEventStakeholders { get; set; }
    }
}

File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.TrainingService.Models/save-training-event_param.ts


import { SaveTrainingEventLocation_Item } from './save-training-event-location_item';
import { KeyActivity_Item } from './key-activity_item';
import { IAA_Item } from './iaa_item';
import { SaveTrainingEventUSPartnerAgency_Item } from './save-training-event-uspartner-agency_item';
import { SaveTrainingEventProjectCode_Item } from './save-training-event-project-code_item';
import { SaveTrainingEventStakeholder_Item } from './save-training-event-stakeholder_item';
import { ISaveTrainingEvent_Param } from './isave-training-event_param';

export class SaveTrainingEvent_Param implements ISaveTrainingEvent_Param {
  
	public TrainingEventID?: number;
	public Name: string = "";
	public NameInLocalLang: string = "";
	public TrainingEventTypeID: number = 0;
	public Justification: string = "";
	public Objectives: string = "";
	public ParticipantProfile: string = "";
	public SpecialRequirements: string = "";
	public ProgramID?: number;
	public KeyActivityID?: number;
	public TrainingUnitID: number = 0;
	public CountryID?: number;
	public PostID?: number;
	public ConsularDistrictID?: number;
	public OrganizerAppUserID?: number;
	public PlannedParticipantCnt?: number;
	public PlannedMissionDirectHireCnt?: number;
	public PlannedNonMissionDirectHireCnt?: number;
	public PlannedMissionOutsourceCnt?: number;
	public PlannedOtherCnt?: number;
	public EstimatedBudget?: number;
	public ActualBudget?: number;
	public EstimatedStudents?: number;
	public FundingSourceID?: number;
	public AuthorizingLawID?: number;
	public Comments: string = "";
	public TrainingEventLocations?: SaveTrainingEventLocation_Item[];
	public KeyActivities?: KeyActivity_Item[];
	public IAAs?: IAA_Item[];
	public TrainingEventUSPartnerAgencies?: SaveTrainingEventUSPartnerAgency_Item[];
	public TrainingEventProjectCodes?: SaveTrainingEventProjectCode_Item[];
	public TrainingEventStakeholders?: SaveTrainingEventStakeholder_Item[];
  
}






